libqtile/widget/generic_poll_text.py

```diff
@@ -92,3 +92,47 @@ def poll(self):
             text = "Can't parse"
 
         return text
+
+    
+class GenPollCommand(base.ThreadPoolText):
+    """A generic text widget to display output from scripts or shell commands"""
+
+    defaults = [
+        ("update_interval", 60,"update time in seconds"),
+        ("cmd", "date", "command line as a string or list of arguments to execute"),
+        ("shell", True, "run command through a shell to enable piping and shell expansion"),
+        ("text_before", "", "additional text to display before command output"),
+        ("text_after", "", "additional text to display after command output"),
+    ]
+
+    def __init__(self, **config):
+        base.ThreadPoolText.__init__(self, "", **config)
+        self.add_defaults(Output.defaults)
+
+    def _configure(self, qtile, bar):
+        base.ThreadPoolText._configure(self, qtile, bar)
+
+        self.text_before = str(self.text_before)
+        self.text_after = str(self.text_after)
+
+        self.add_callbacks({
+            "Button1": self.cmd_force_update,
+        })
+
+    def poll(self):
+        self.run_process()
+        return self.get_output()
+
+    def run_process(self):
+        self.completed_process = subprocess.run(
+            self.cmd,
+            capture_output=True,
+            text=True,
+            shell=self.shell,
+        )
+        return self.completed_process
+
+    def get_output(self):
+        output = self.completed_process.stdout.strip()
+        output = self.text_before + output + self.text_after
+        return str(output)
```

Comment:
Doesn't `text=True` mean you get a string object already? If so, you could just `return output`.